src/java/com/sarality/app/view/list/ListViewItemActionPerformer.java

```diff
@@ -0,0 +1,93 @@
+package com.sarality.app.view.list;
+
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+
+import com.sarality.app.view.action.Action;
+import com.sarality.app.view.action.TriggerType;
+
+/**
+ * Performs the Action when an event on an Item in a ListView is triggered.
+ * 
+ * @author abhideep@ (Abhideep Singh)
+ */
+class ListViewItemActionPerformer implements AdapterView.OnItemClickListener, AdapterView.OnItemLongClickListener,
+    View.OnClickListener, View.OnLongClickListener {
+
+  private final static String LIST_POSITION_TAG = "ListPosition";
```

Comment:
We have been doing this with Tags(Key = view.getId).  

now that we are going to set up a new set of resourceIds for ActionPerformers , we should have one set aside to save the data also.  It seems cleaner than passing it in the constructor.